kong/pdk/log.lua

```diff
@@ -809,7 +808,7 @@ do
         service = ctx.service,
         consumer = ctx.authenticated_consumer,
         client_ip = var.remote_addr,
-        started_at = ctx.KONG_PROCESSING_START or (req.start_time() * 1000)
+        started_at = okong.request.get_start_time(),
```

Comment:
I'm not familiar with how the PDK is laid out and the reasons behind it, but if we are going to expose more timestamps and latencies from the PDK, having a dedicated module to do that may be better putting things into the `request` module.
`kong.telemetry`, `kong.time` or something along those lines may be a good option.

Reply:
Oh boy I would _love_ to move all of the `ctx.KONG_*_{START,ENDED_AT,LATENCY,TIME}` read/write/calculate stuff into its own module (with a corresponding PDK interface). I've wondered many a time why this hasn't been done before; it seems like low-hanging fruit (maybe there's an angle to it that I'm missing).

Reply:
>  I've wondered many a time why this hasn't been done before; it seems like low-hanging fruit (maybe there's an angle to it that I'm missing).

I can think of a few different reasons why this is the case. But yes, it is low-hanging fruit, and, there are plenty of those in the codebase. I'll say that it takes skill and a _bit_ of thinking to recognize those.

Reply:
Resolving this convo (for now) but keeping these ideas in mind. It feels a little late in the release cycle for me to start touching lots of stuff in `kong/init.lua`, and it should be trivial to maintain the backwards compatibility of this new `kong.request.get_start_time` function if/when I can revisit all of this.